include/ozo/failover/role_based.h

```diff
@@ -591,14 +590,20 @@ class role_based_strategy : public options_factory_base<role_based_strategy<Opti
 
 /**
  * Try to perform an operation on first role with fallback to next items of
- * specified sequence in case of recoverable error.
+ * specified sequence which should recover an error.
+ * E.g., in case of sequence consists of <i>role1</i>, <i>role2</i>, <i>role3</i> and <i>role4</i>.
+ * If <i>role1</i> cause an error which could be recovered with <i>role3</i> and can not be
```

Comment:
don't we want to add example to this case?

Reply:
Yes we do, but I'm not ready to provide good example right now :(